test/SILOptimizer/split_critical_edges.sil

```diff
@@ -90,9 +90,7 @@ lookup1:
   %3 = alloc_box $<τ_0_0> { var τ_0_0 } <Optional<() -> ()>>
   %4 = load %1a : $*AnyObject
   strong_retain %4 : $AnyObject
-  %6 = open_existential_ref %4 : $AnyObject to $@opened("01234567-89ab-cdef-0123-000000000000") AnyObject
-  %7 = unchecked_ref_cast %6 : $@opened("01234567-89ab-cdef-0123-000000000000") AnyObject to $Builtin.UnknownObject
-  dynamic_method_br %7 : $Builtin.UnknownObject, #X.f!1.foreign, bb1, bb2
+  dynamic_method_br %4 : $AnyObject, #X.f!1.foreign, bb1, bb2
```

Comment:
This one doesn't quite match what we emit today but I figured the test relied on the two branches having the same type.